src/widgets/chart.rs

```diff
@@ -560,7 +560,7 @@ impl<'a> Widget for Chart<'a> {
 
         if let Some((x, y)) = layout.title_x {
             let title = self.x_axis.title.unwrap();
-            let width = graph_area.right().saturating_sub(x);
+            let width = title.width() as u16;
```

Comment:
How does this act when title.width is > buffer width? Can you please add a test to ensure that this works fine without panicking?

Reply:
The title is hidden when it's too long. It's checked [here](https://github.com/ratatui-org/ratatui/blob/e098731d6c1a68a0319d544301ac91cf2d05ccb2/src/widgets/chart.rs#L302C13-L302C13).

Reply:
Thanks for implementing this. The code that sets up the invariant is quite a long way (160 lines prior) to the code that relies on it.